server/sonar-server/src/main/java/org/sonar/server/ws/WsParameterBuilder.java

```diff
@@ -45,6 +45,13 @@ private WsParameterBuilder() {
       .setPossibleValues(getRootQualifiers(context.getResourceTypes()));
   }
 
+  public static WebService.NewParam createRootQualifiersParameter(WebService.NewAction action, QualifierParameterContext context) {
```

Comment:
As discussed, it would have been best to refactor WsParameterBuilder.java to use injection for I18n and ResourceTypes resourceTypes instead of using this strange context object.

Reply:
indeed, I haven't taken the time to do it

Reply:
No pb